pkg/apis/networking/v1alpha1/certificate_types.go

```diff
@@ -93,4 +94,27 @@ type CertificateStatus struct {
 	// by this resource in spec.secretName.
 	// +optional
 	NotAfter *metav1.Time `json:"notAfter,omitempty"`
+
+	// HTTP01Challenges is a list of HTTP01 challenges that need to be fulfilled
+	// in order to get the TLS certificate..
+	HTTP01Challenges []HTTP01Challenge `json:"http01Challenges,omitempty"`
+}
+
+// HTTP01Challenge defines the status of a HTTP01 challenge that a certificate needs
+// to fulfill.
+type HTTP01Challenge struct {
+	// DNSName is the DNS name to be verified by this HTTP01 challenge.
+	DNSName string `json:"dnsName,omitempty"`
```

Comment:
I wonder if we should combine this and Path into an `apis.URL` that the challenge is expected to serve on.

Reply:
SGTM. Updated to use URL.